components/form-builder/app/edit/PanelActions.tsx

```diff
@@ -18,67 +18,62 @@ import { usePanelActions } from "@components/form-builder/hooks";
 import { ElementDialog } from "./elements/element-dialog/ElementDialog";
 import { PanelActionsButton } from "./PanelActionsButton";
 
-export interface RenderMoreFunc {
-  ({ item, moreButton }: { item: FormElementWithIndex; moreButton: JSX.Element | undefined }):
-    | React.ReactElement
-    | string
-    | undefined;
-}
-
 export const PanelActions = ({
-  item,
-  subIndex,
-  renderMoreButton,
+  isFirstItem,
+  isLastItem,
+  totalItems,
+  isSubPanel,
+  moreButtonRenderer,
   handleAdd,
   handleRemove,
   handleMoveUp,
   handleMoveDown,
   handleDuplicate,
   filterElements,
-  elements,
 }: {
-  item: FormElementWithIndex;
-  subIndex?: number;
-  renderMoreButton: RenderMoreFunc;
-  handleAdd: (index: number, type?: FormElementTypes) => void;
+  isFirstItem: boolean;
+  isLastItem: boolean;
+  totalItems: number;
+  isSubPanel?: boolean;
+  moreButtonRenderer?: RenderMoreFunc;
+  handleAdd: (type?: FormElementTypes) => void;
   handleRemove: () => void;
   handleMoveUp: () => void;
   handleMoveDown: () => void;
   handleDuplicate: () => void;
   filterElements?: ElementOptionsFilter;
-  elements: FormElement[];
 }) => {
   const { t, i18n } = useTranslation("form-builder");
 
   const isInit = useRef(false);
-  const isLastItem = item.index === elements.length - 1;
-  const isFirstItem = item.index === 0;
-  const isRichText = item.type == "richText";
-  const isSubElement = subIndex !== -1 && subIndex !== undefined;
   const lang = i18n.language;
+  const hasMoreButton = moreButtonRenderer ? true : false;
+  const getPanelButtons = ({ hasMoreButton }: { hasMoreButton: boolean }) => {
+    const panelButtons = [
```

Comment:
What do you think about something like this to keep more inline with the original code vs the pushes.  I found the original where we had one specific array for each panel type clearer to read.

```javascript
let test = false;

const items = [
  { id: 1, txt: "moveUp" },
  { id: 2, txt: "moveDown" },
  test && { id: 3, txt: "removeFromSet" }
].filter(Boolean);

console.log(items);
```